### ISSUE TITLE
add Landsat Collection 1 cloud mask based on the quality assessment band

### ISSUE THREAD

**Issue by ircwaves** [Applied-GeoSolutions/gips#341]
Documentation on the QA Band: https://landsat.usgs.gov/collectionqualityband

USGS Ref impl: https://github.com/USGS-EROS/landsat-ldope-tools

For a first cut, the mask should be 1 everywhere that is neither cloud nor cloud shadow.

**Comment by ircwaves** [comment]
Cloudy day: https://earthexplorer.usgs.gov/download/12864/LC80120302017181LGN00/STANDARD/EE

Better scene browser: https://libra.developmentseed.org/

**Comment by ags-tolson** [comment]
So I beat my head against this for awhile, but I can't seem to make `binmask` work as an alternative to `get_bit` in that branch; I kept getting opaque exceptions that didn't make much sense.  Here's a version of the cloudmask stanza that broke:

```
                    elif val[0] == 'cloudmask':
                        qaimg = self._readqa()
                        npqa = qaimg.Read() # transmogrify into numpy array
                        np_cloudmask = binmask(npqa, 4) & binmask(npqa, 6) | binmask(npqa, 8) # seems to work
                        imgout = gippy.GeoImage(fname, img, gippy.GDT_Byte, 1) # only one layer
                        verbose_out("writing " + fname, 2)
                        imgout.SetBandName('Cloud Mask', 1)
                        imgout[0].Write(np_cloudmask.astype('byte'))  # breakage here
 ```

I also tried `'int8'` and that didn't work either.  Here's the error I'd get:

```
Error creating product cloudmask for LC08_L1TP_012030_20170630_20170715_01_T1.tar:
Traceback (most recent call last):
  File "/home/tolson/src/gips/gips/utils.py", line 475, in cli_error_handler
    yield
  File "/home/tolson/src/gips/gips/data/landsat/landsat.py", line 990, in process
    imgout[0].Write(np_cloudmask.astype('byte'))
  File "/home/tolson/src/gips/.dhve/local/lib/python2.7/site-packages/gippy/algorithms.py", line 3856, in Write
    return _algorithms.GeoRaster_Write(self, *args)
RuntimeError: std::exception
```

**Comment by ags-tolson** [comment]
The better news is that the cloudmask generated for the day in question looks vaguely correct; should I hand this to someone with science chops who can verify it?  Files:  goo.gl/jojbij

**Comment by ags-tolson** [comment]
When development resumes, here is the branch:

https://github.com/Applied-GeoSolutions/gips/tree/341-cloud

**Comment by ags-tolson** [comment]
fix in PR #349
